[Ramapriyan Srivatsan Purisai Devarajan] iP

ramapriyan912001 opened the pull request:
### Duke frees your mind of having to remember things you need to do. It's,

• text-based
• easy to learn
• ~~FAST~~ SUPER FAST to use

All you need to do is,

- [x] download it from here.
- [x] double-click it.
- [ ] add your tasks.
- [ ] let it manage your tasks for you 😉
- [ ] And it is FREE!

### Features:
1.  Managing tasks
2.  Managing deadlines (coming soon)
3.  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

A link to my orbital app frontend: https://github.com/ramapriyan912001/gobble-app-frontend

As Someone once famously said:
> Don't be feelings droppin'
> 'Cuz u a goat
> Ain't got time for no trippin'

I like to `<code>` coz I'm a codeine to the history books yo.

Commit:
Add Gradle support

ramapriyan912001 committed:
Level-1

ramapriyan912001 committed:
Level-2

ramapriyan912001 committed:
Level-3

ramapriyan912001 committed:
Level-4

ramapriyan912001 committed:
Level-5 functioning but automated testing not working out, issue posted to github

ramapriyan912001 committed:
Previous commit erroneous due to the fact that all changes were not staged, Level-5 now done

ramapriyan912001 committed:
Level-6 delete functionality complete

ramapriyan912001 committed:
minor changes to try to get automtated testing working - still fails

ramapriyan912001 committed:
Automated Text UI Testing working

ramapriyan912001 committed:
Add delete test

ramapriyan912001 committed:
Level-6 complete with Javadocs

ramapriyan912001 committed:
enums commented in

ramapriyan912001 committed:
Level-7 Complete

ramapriyan912001 committed:
Level-8

ramapriyan912001 committed:
Modularised code further

ramapriyan912001 committed:
Enabled Find functionality

ramapriyan912001 committed:
Added Javadocs comments

ramapriyan912001 committed:
Updated javadoc comments

ramapriyan912001 committed:
Enforce Coding Standard

ramapriyan912001 committed:
Update filter

ramapriyan912001 committed:
Merge branch 'A-CodingStandard'
Merging coding standard branch to master

ramapriyan912001 committed:
Merge branch 'Level-9'
Merging Level-9 branch to master.

ramapriyan912001 committed:
Added 2 Junit test methods

ramapriyan912001 committed:
Package into duke package

ramapriyan912001 committed:
Enforce coding standard further

ramapriyan912001 committed:
Complete making JAR

ramapriyan912001 committed:
Minor edit to formatting

ramapriyan912001 committed:
Comment Javadocs on Duke Exceptions class

ramapriyan912001 committed:
Merge pull request #2 from ramapriyan912001/Level-8

Comment Javadocs on Duke Exceptions class

ramapriyan912001 committed:
Fix consistency issues tasklist and taskmanager

ramapriyan912001 committed:
Merge pull request #4 from ramapriyan912001/Level-7

Level 7

ramapriyan912001 committed:
Merge branch 'master' into add-gradle-support

ramapriyan912001 committed:
Gradle files

ramapriyan912001 committed:
Merge branch 'master' of https://github.com/ramapriyan912001/ip

ramapriyan912001 committed:
Complete setup

ramapriyan912001 committed:
Merge branch 'add-gradle-support' of https://github.com/ramapriyan912001/ip

ramapriyan912001 committed:
Fix checkstyle issues

ramapriyan912001 committed:
Set up Duke javafx UI but does not run due to error

ramapriyan912001 committed:
Iteration 2 does not work

ramapriyan912001 committed:
Complete Iteration 3

ramapriyan912001 committed:
Add fxml files

ramapriyan912001 committed:
Integrate chatbot with javafx ui

ramapriyan912001 committed:
Fix checkstyle

wangpalex commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,165 @@
+     * @param input The string of input command.
+     * @throws DukeException Exceptions specific to this chatbot.
+     */
+    public String interpretInput(String input) throws DukeException {
```
It could be better to extract out some of the processing logic into other methods to make this methed shorter and more readable.

ramapriyan912001 replied:
Thank you for the feedback Alex, will look to resolve this asap.

wangpalex reviewed (commented):
LGTM. Great job on keeping code clear and readable!

mingyi456 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,29 @@
+
+    private TaskList taskManager = new TaskList();
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
I think this does not follow the JavaDoc conventions because it lacks a @return parameter, but I believe this is something might have missed by accident, because many other methods have the proper JavaDocs documentation.

ramapriyan912001 replied:
Thank you for the feedback, will look to resolve this asap

mingyi456 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,16 @@
+package duke;
+
+/**
+ * Class of Duke Exceptions.
```
This JavaDocs seems a little trivial to me, maybe make it more substantial in the future?

ramapriyan912001 replied:
Sounds good, thank you.

mingyi456 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    /**
+     * String reflecting task itself.
```
I think "String reflecting this task" would fit the JavaDocs conventions better?

mingyi456 commented on src/main/java/duke/Type.java:
```diff
@@ -0,0 +1,39 @@
+/**
+ * Enum class for types of events.
+ */
+public enum Type {
```
I think this naming might be a bit confusing if I have not seen the rest of your code, because you are referring to the type of `Task`, so maybe `TaskType` would be more suitable?

mingyi456 commented on src/main/java/duke/Type.java:
```diff
@@ -0,0 +1,39 @@
+     * @param type Type enum.
+     * @return String representation of type.
+     */
+    public static String typeString(Type type) {
+        if (type == Type.EVENT) {
+            return "E";
+        } else if (type == Type.TODO) {
+            return "T";
+        } else if (type == Type.DEADLINE) {
+            return "D";
+        } else {
+            throw new NullPointerException();
+        }
+    }
+
+    public static String getConnector(Type type) {
+        if (type == Type.EVENT) {
+            return "at";
+        } else if (type == Type.TODO) {
+            return "";
+        } else if (type == Type.DEADLINE) {
+            return "by";
+        } else {
+            throw new NullPointerException();
+        }
+    }
```
I feel this is a very good use of `Enums` which cleans up a lot of code in other areas, and I should have done this for my own project too.

mingyi456 reviewed (approved):
Overall, I think this is a really well done PR, and it is very difficult for me to spot any flaws. Please note that anything I did manage to point out is very trivial compared to the overall high quality of everything else. I am really impressed that you have already implemented and merged your JavaFX code, unlike me who is still behind the schedule.

Edit: Oh no, I accidentally clicked approve instead of comment. Is there any way to change this?

ramapriyan912001 committed:
Insert assert statement

An assert statement to check whether command is valid.
Previously handled by an exception class.

ramapriyan912001 committed:
Merge pull request #5 from ramapriyan912001/A-Assertions

A-Assertions

ramapriyan912001 committed:
Add sort functionality

The chatbot can now sort events and deadlines based on the time assigned to them.

ramapriyan912001 committed:
Merge branch 'master' into C-Sort

ramapriyan912001 committed:
Merge pull request #6 from ramapriyan912001/C-Sort

Add sort functionality

ramapriyan912001 committed:
Modularise code to adhere to code quality standards

Functions dealing with the interpreting of the user input were moved into the parser class to improve readability and organisation of the code

ramapriyan912001 committed:
Merge pull request #7 from ramapriyan912001/A-CodeQuality

Modularise code to adhere to code quality standards

ramapriyan912001 committed:
Refractor Test class

ramapriyan912001 committed:
Merge C-Sort

ramapriyan912001 committed:
Fix testing issues

ramapriyan912001 committed:
Fix minor issues

ramapriyan912001 committed:
Add user guide

ramapriyan912001 committed:
Fix UI screenshot and complete user guide

ramapriyan912001 committed:
Rename UI screenshot file

ramapriyan912001 committed:
Merge branch 'master' of https://github.com/ramapriyan912001/ip

ramapriyan912001 committed:
Improve User Guide

ramapriyan912001 committed:
Enforce SLAP

ramapriyan912001 committed:
Fix minor syntax bugs

ramapriyan912001 committed:
Fix Checkstyle issue with order of imports

ramapriyan912001 committed:
Add screenshot of chatbot to README

ramapriyan912001 committed:
Fix assertion error bug

ramapriyan912001 committed:
Remove hardcoded file paths

ramapriyan912001 committed:
Add EmptyEventException

ramapriyan912001 committed:
Change Gradle version

ramapriyan912001 committed:
Change gradle version details
;

ramapriyan912001 committed:
Add function to create file if file does not exist

ramapriyan912001 committed:
Alter dialogbox fxml file to show more than 3 tasks for list command

ramapriyan912001 committed:
Fix font issue

ramapriyan912001 committed:
Add better images for duke and self

ramapriyan912001 committed:
Fix font issue on some Macs

ramapriyan912001 committed:
Fix dialogbox styling

ramapriyan912001 committed:
Fixes for CS2103bot suggestions
